.github/workflows/ci.yml

```
@@ -0,0 +1,143 @@
+      spack-packages-ref:
+        required: false
+        type: string
+        default: main
+        description: |
+          The branch, tag, or commit SHA of the access-nri/spack-packages repository to use.
+          For example: main, 2025.03.0, 7ey2uy2.
+      spack-ref:
+        required: false
+        type: string
+        default: main
+        description: |
+          The branch, tag, or commit SHA of the access-nri/spack repository to use.
+          For example: develop, releases/0.22, 7ey2uy2.
+      pytest-test-markers:
+        required: false
+        type: string
+        description: |
+          A string of pytest markers to use to filter tests in the caller model component repository's .github/build/tests directory.
+          For example: "not slow and not mpi".
+      allow-ssh-into-spack-install:
+        required: false
+        type: boolean
+        default: false
+        description: |
+          Enable the actor of the workflow to SSH into the container where the spack packages have been installed.
+          This is useful for gathering post-install information before the container is destroyed.
+          This will also make the workflow wait until the actor SSHs into the container, or it times out, before continuing.
+    outputs:
+      spec-concretization-graph:
+        value: ${{jobs.spack-install-and-test.outputs.spec-concretization-graph }}
+        description: |
+          A visual representation of the dependencies and constraints of the spack manifest file installed.
+      # test-artifact-url:
+      #   value: ${{}}
+      #   description: |
+      #     The URL of the pytest result artifact.
+jobs:
+  spack-install-and-test:
+    name: Install and Test
+    runs-on: self-hosted  # FIXME: This will be the group of self-hosted runners for build-ci
+    container:
+      image: ghcr.io/access-nri/spack-ci:runner-rocky
+      credentials:
+        username: ${{ github.actor }}
+        password: ${{ secrets.GHCR_PASSWORD }}  # FIXME: Set appropriate PAT
+      volumes:
+        - upstream:/opt/upstream:ro  # FIXME: Validate connection via container -> runner -> upstream pod
+    outputs:
+      spec-concretization-graph: ${{ steps.install.outputs.spec }}
+      # test-artifact-url: ${{ steps.test.outputs.artifact-url }}
+    steps:
+      - name: Export environment variables into GitHub Actions format
+        # Environment variables inside containers are not accessible in the `env` context,
+        # a context which would be used in future conditional steps.
+        # This step exports important container environment variables as outputs.
+        id: env
+        run: |
+          echo "SPACK_ROOT=$SPACK_ROOT" >> $GITHUB_OUTPUT
+          echo "SPACK_VERSION=$SPACK_VERSION" >> $GITHUB_OUTPUT
+          echo "SPACK_CONFIG_REPO_VERSION=$SPACK_CONFIG_REPO_VERSION" >> $GITHUB_OUTPUT
+          echo "SPACK_PACKAGES_REPO_VERSION=$SPACK_PACKAGES_REPO_VERSION" >> $GITHUB_OUTPUT
```

Might be good to have this as an action to make the logic clearer for dummy users like me.

---

I've realised that we would need to create a new output for each environment variable that we add to the potential action. I think that turning this into an action may actually lead to more lines of code that need to be changed to add a given environment variable (eg. add a variable as an action input, add an output in the action of the variable, etc. )

---

Fair enough.

Maybe just a loop
```suggestion
          for VAR in "SPACK_ROOT"  "SPACK_VERSION" "SPACK_CONFIG" "SPACK_PACKAGES"; do
              echo "$VAR=${!VAR}" >> $GITHUB_OUTPUT
          done
```

or to export all `SPACK_` related variables

```suggestion
          for VAR in ${!SPACK_*}; do
              echo "$VAR=${!VAR}" >> $GITHUB_OUTPUT
          done
```
(from https://stackoverflow.com/a/25765828)

---

Cool! Done in https://github.com/ACCESS-NRI/build-ci/pull/200/commits/457dbc5605d43a4fae96710d5e8bad51288711fd